### PR TITLE
changed hyperlink to supported model doc page

### PR DESCRIPTION
**Pull request by pagezyhf**
# What does this PR do?

changed hyperlink to supported-architectures

## Before submitting
- [ X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you make sure to update the documentation with your changes?
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by dacorvo — COMMENTED** [review]
I don't understand the change, since the link was actually correct.

File: docs/source/index.mdx [hunk 1]
```diff
@@ -18,7 +18,7 @@ limitations under the License.
 
 🤗 Optimum Neuron is the interface between the 🤗 Transformers library and AWS Accelerators including [AWS Trainium](https://aws.amazon.com/machine-learning/trainium/?nc1=h_ls) and [AWS Inferentia](https://aws.amazon.com/machine-learning/inferentia/?nc1=h_ls).
 It provides a set of tools enabling easy model loading, training and inference on single- and multi-Accelerator settings for different downstream tasks.
-The list of officially validated models and tasks is available [here](https://huggingface.co/docs/optimum-neuron/package_reference/configuration#supported-architectures).
+The list of officially validated models and tasks is available [here](https://huggingface.co/docs/optimum-neuron/package_reference/supported-architectures).
```
Comment (dacorvo) [inline docs/source/index.mdx:21]:
```suggestion
The list of officially validated models and tasks is available [here](https://huggingface.co/docs/optimum-neuron/package_reference/supported-models).
```

**Review by dacorvo — COMMENTED** [review]
Ok to change the link to avoid the indirection, but there was a typo.
Otherwise LGTM.

**Comment by pagezyhf** [comment]
Thank you @dacorvo for picking it up and catching the typo!

**Pull request closed without merging**